docs/standard/datetime/datetime-support-in-system-text-json.md

```diff
@@ -0,0 +1,194 @@
+
+If you can't determine the formats of your input <xref:System.DateTime> or <xref:System.DateTimeOffset> text representations, you can use the
+`DateTime(Offset).Parse` method in your converter read logic. This allows you to use .NET's extensive support for parsing various
+<xref:System.DateTime> and <xref:System.DateTimeOffset> text formats, including non-ISO 8601 strings and ISO 8601 formats that don't conform to
+the extended ISO 8601-1:2019 profile. This approach is significantly less performant than using the serializer's native implementation.
+
+For serializing, you can use the `DateTime(Offset).ToString` method in your converter write logic. This allows you to write <xref:System.DateTime>
+and <xref:System.DateTimeOffset> values using any of the
+[standard date and time formats](https://docs.microsoft.com/dotnet/standard/base-types/standard-date-and-time-format-strings).
+This is also significantly less performant than using the serializer's native implementation.
+
+[!code-csharp[example-showing-datetime-parse](~/samples/snippets/standard/datetime/json/datetime-converter-examples/example1/Program.cs)]
+
+> [!NOTE]
+> When implementing JsonConverter<DateTime>, the `typeToConvert` parameter will always be typeof(DateTime). The parameter is useful for polymorphic cases and when using generics to get typeof(T) in a performant way.
+
+### Using <xref:System.Buffers.Text.Utf8Parser> and <xref:System.Buffers.Text.Utf8Formatter>
+
+You can used fast UTF-8-based parsing and formatting methods in your converter logic if your input <xref:System.DateTime> or <xref:System.DateTimeOffset>
+text representations are compliant with one of the "R", "l", "O", or "G"
+[standard date and time Format Strings](https://docs.microsoft.com/dotnet/standard/base-types/standard-date-and-time-format-strings),
+or you want to write according to one of these formats. This is much faster than using `DateTime(Offset).Parse` and `DateTime(Offset).ToString`
+
+This example shows a custom converter that serializes and deserializes <xref:System.DateTime> values according to
+[the "R" standard format](https://docs.microsoft.com/dotnet/standard/base-types/standard-date-and-time-format-strings#the-rfc1123-r-r-format-specifier):
+
+[!code-csharp[example-showing-utf8-parser-and-formatter](~/samples/snippets/standard/datetime/json/datetime-converter-examples/example2/Program.cs)]
+
+> [!NOTE]
+> The "R" standard format will always be 29 characters long.
+
+### Using `DateTime(Offset).Parse` as a fallback to the serializer's native parsing
+
+If you generally expect your input <xref:System.DateTime> or <xref:System.DateTimeOffset> data to conform to the extended ISO 8601-1:2019 profile,
+you can use the serializer's native parsing logic. You can also implement a fallback mechanism just in case.
+This example shows that, after failing to parse a <xref:System.DateTime> text representation using <xref:System.Text.Json.Utf8JsonReader.TryGetDateTime(System.DateTime)>,
```

Comment:
I'm not sure why the ref to <xref:System.Text.Json.Utf8JsonReader.TryGetDateTime(System.DateTime)> isn't working.

Reply:
Try \<xref:System.Text.Json.Utf8JsonReader.TryGetDateTime(System.DateTime@)> to indicate the DateTime value is passed by reference.

Reply:
This worked, thank you.